temporian/core/event_set_ops.py

```diff
@@ -2475,3 +2475,84 @@ def unique_timestamps(self: EventSetOrNode) -> EventSetOrNode:
         from temporian.core.operators.unique_timestamps import unique_timestamps
 
         return unique_timestamps(self)
+
+    def where(
+        self: EventSetOrNode,
+        on_true: Union[EventSetOrNode, Any],
```

Comment:
Should we create an alias for `Union[int, float, str, bool]`? we have `T_SCALAR` defined in event_set_ops which is only (int, float), but int that same file in `__ne__`'s def we join it with `(bool, str)` too. Maybe this definition could improve the signature's legibility

Reply:
I'm not sure about this. `T_SCALAR` is actually a tuple, we're not using it in typehints.

This would make sense if we define types in `typing.py` like `PythonValue` and `PythonScalarValue` and use it consistently in the event_set_ops types.
But I think we should discuss it previously, I would not add it in this PR.